packages/compass-collection/src/components/collection/collection.tsx

```diff
@@ -122,6 +122,22 @@ const Collection: React.FunctionComponent<CollectionProps> = ({
     [id, activeSubTab, localAppRegistry, globalAppRegistry, changeActiveSubTab]
   );
 
+  useMemo(() => {
+    const indexesTabId = tabs.indexOf('Indexes');
+    const onOpenCreateIndexEvent = onSubTabClicked.bind(
+      null,
+      indexesTabId,
+      tabs[indexesTabId]
+    );
+    localAppRegistry.on('open-create-index-modal', onOpenCreateIndexEvent);
+    return () => {
```

Comment:
Is this supposed to be a `useEffect`…? Otherwise I’m not quite sure what the return value here means

Reply:
Yes, good catch, thanks